Automattic/wp-calypso
# Customize Auth flow for Blaze Pro

Related to A8C-DSP's issue `1439`.

## Proposed Changes
* Adjust common styles for the case of BlazePro (base on WooCommerce's Auth flow implementation).

## Why are these changes being made?
* We need to customize WPCOM Authentication flow for BlazePro (in the same way it's implemented for WooCommerce)

Designs in [Figma](https://www.figma.com/design/0YQJMNMOVVOMqzCWufls0l/BlazePro-(Self-Served-for-Advertisers)?node-id=7096-11442&t=jNKJPuFzcHDxg3TE-0).

## Testing Instructions
Despite the PR `1520` from A8C-DSP repo is the starting point for running Blaze Pro, we can test changes in Calypso with the use of the following script that generates links for Login and Sign Up (it should be run with `node`).

```js
function generateState() {
  const crypto = require('crypto');
  const hmac = crypto.createHmac('sha256', 'secret' );
  const data = hmac.update(new Date().getTime().toString());
  const state = data.digest('hex');
  return state;
}

const redirectUri = 'http://blaze.pro:3005/auth/connected&blog_id=0&wpcom_connect=1&calypso_env=development&scope=global&from-calypso=1';
const getLoginLink = () =>
	`https://public-api.wordpress.com/oauth2/authorize?response_type=code&client_id=92099&state=${generateState()}&redirect_uri=${encodeURIComponent(redirectUri)}`;

const signInLink = `http://calypso.localhost:3000/log-in?client_id=92099&redirect_to=${encodeURIComponent(getLoginLink())}`;

const signUpLink = `http://calypso.localhost:3000/start/wpcc/oauth2-user?oauth2_client_id=92099&oauth2_redirect=${encodeURIComponent(getLoginLink())}`;

console.log('Login URL:');
console.log(signInLink);
console.log('');
console.log('Sign Up URL:');
console.log(signUpLink);
```

**Caveat 1**: I noticed that for Google Chrome uses "DNS over HTTPS", so for me it's impossible to use it for sandboxing. So, I used Firefox for testing.

**Caveat 2**: I was not able to have authentication to be working locally. I assume that as long as it will be merged it should work properly due to having `wordpress.com` instead of `calypso.localhost:3000`.

### Login Flow
- Sandbox WPCOM,
- Apply D149804-code,
- Use the script to generate a URL,
- Run Calypso locally,
- Open a URL in your browser (like `https://public-api.wordpress.com/oauth2/authorize?response_type=code&client_id=92099&...`),
- Ensure you see Login page similar to image below:

<img width="1343" alt="image" src="https://github.com/Automattic/wp-calypso/assets/115007291/7b9178db-b6af-43e2-81c9-57d3d9abc704">

- Use your WPCOM credentials to continue
- Due to inability to authenticate locally, as a sign of being authenticated in WPCOM we should see a screen of already authenticated on WPCOM's side:

<img width="1341" alt="image" src="https://github.com/Automattic/wp-calypso/assets/115007291/aaf85ec5-0713-43ac-99bc-0ca762e90e08">

Short demo:

https://github.com/Automattic/wp-calypso/assets/115007291/76bcb0d0-553e-411f-b8df-beb5ad3c26c7

### Sign Up Flow
(Continue testing)
- Clear browser cookies,
- Adjust code ([source](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/main.jsx#L841)) in `client/signup/main.jsx:850` in order to show Processing Screen from:
```js
{ this.state.shouldShowLoadingScreen ? (
```

to:

```js
{ true || this.state.shouldShowLoadingScreen ? (
```

- Get a URL for Sign Up from the script,
- Open a URL in your browser (like `http://calypso.localhost:3000/start/wpcc/oauth2-user?oauth2_client_id=92099&...`),

![processing-screen](https://github.com/Automattic/wp-calypso/assets/115007291/c299b9bd-625b-436c-a5c0-9f4b2ae40cb7)

- Remove code adjustment you've done before
- Refresh the page and make sure you see Sign Up form

<img width="1340" alt="image" src="https://github.com/Automattic/wp-calypso/assets/115007291/fe7b6557-8f31-4f08-8753-f7d1d4fa892f">

Short demo:

https://github.com/Automattic/wp-calypso/assets/115007291/a9ea1d2f-85ec-4ff7-8f5d-fa0c58f6a0a3

## Pre-merge Checklist
